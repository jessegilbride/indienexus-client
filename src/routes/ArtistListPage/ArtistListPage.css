.artist-list-banner {
  background: url('../../images/all-artists-banner.jpg');
  height: 280px;
}

.artist-list {
  padding-bottom: 2rem;
}