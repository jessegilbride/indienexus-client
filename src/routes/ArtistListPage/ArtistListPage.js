import React, { Component, Fragment } from 'react';
import ArtistListContext from '../../contexts/ArtistListContext';
import ArtistApiService from '../../services/artist-api-service';
import ArtistListItem from '../../components/ArtistListItem/ArtistListItem';
import './ArtistListPage.css';

export default class ArtistListPage extends Component {
  static contextType = ArtistListContext;

  componentDidMount() {
    this.context.clearError();
    ArtistApiService.getArtists()
      .then(this.context.setArtistList)
      .catch(error => {
        console.log(error)
        return this.context.setError
      });
  }

  renderArtists() {
    const { artistList = [] } = this.context;
    return artistList.map((artist) => <ArtistListItem key={artist.id} artist={artist} />);
  }

  render() {
    const { error } = this.context;
    return (
      <Fragment>
        <section className="page-width-container page-top-banner artist-list-banner">
          <h2 className='page-heading'>All Artists</h2>
        </section>
        <section className="artist-list page-width-container">
          {error ? (
            <p className='red'>There was an error, try again</p>
          ) : (
            this.renderArtists()
          )}
        </section>
      </Fragment>
    );
  }
}
