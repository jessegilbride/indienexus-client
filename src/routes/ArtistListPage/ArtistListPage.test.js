import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import ArtistListPage from './ArtistListPage';

describe('ArtistListPage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <MemoryRouter><ArtistListPage /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <MemoryRouter><ArtistListPage name="ArtistListPage" /></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});
