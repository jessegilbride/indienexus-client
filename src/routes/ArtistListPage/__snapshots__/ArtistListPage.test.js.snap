// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistListPage component renders the UI as expected 1`] = `
Array [
  <section
    className="page-width-container page-top-banner artist-list-banner"
  >
    <h2
      className="page-heading"
    >
      List of All Artists
    </h2>
  </section>,
  <section
    className="artist-list page-width-container"
  />,
]
`;
